src/utils.ts

```diff
@@ -10,14 +10,54 @@ import {execSync} from 'child_process';
 import * as mkdirp from 'mkdirp';
 const isWsl = require('is-wsl');
 
-export function defaults<T>(val: T | undefined, def: T): T {
+export const enum LaunchErrorCodes {
+  ERR_LAUNCHER_PATH_NOT_SET = 'ERR_LAUNCHER_PATH_NOT_SET',
+  ERR_LAUNCHER_INVALID_USER_DATA_DIRECTORY = 'ERR_LAUNCHER_INVALID_USER_DATA_DIRECTORY',
+  ERR_LAUNCHER_UNSUPPORTED_PLATFORM = 'ERR_LAUNCHER_UNSUPPORTED_PLATFORM',
+  ERR_LAUNCHER_NOT_INSTALLED = 'ERR_LAUNCHER_NOT_INSTALLED',
+}
+
+export function defaults<T>(val: T|undefined, def: T): T {
   return typeof val === 'undefined' ? def : val;
 }
 
 export async function delay(time: number) {
   return new Promise(resolve => setTimeout(resolve, time));
 }
 
+export class LauncherError extends Error {
+  constructor(public message: string = 'Unexpected error', public code?: string) {
+    super();
+    this.message = message;
```

Comment:
since you now have the public attr on the arguments in the constructor it means that you can delete
lines 31 though 34

here is a good explainer
https://kendaleiv.com/typescript-constructor-assignment-public-and-private-keywords/

Reply:
@samccone Oh wow, I didn't know this. Fixed in 777aa72.